### PR TITLE
feat(cmd): Introduce centralized command registry and metadata, docs generation

### LINKED ISSUE #106: [Feature]: Centralized Command Registry
## Related Problem

Currently, command information is maintained across multiple locations, causing the following issues:

1. **Duplicate Management of Command Data**

   * `cmd/interactive.go`: `commands` array
   * `cmd/templates/help.go`: main command list
   * `cmd/cmd.go`: routing logic
     Adding a new command requires updating all these places individually.

2. **Difficulty in Maintaining Consistency**

   * Command descriptions and usage may differ between files.
   * Easy to forget to update one of the locations when adding/removing commands.

## Proposed Solution

Introduce a new `command` package and implement a centralized command registry:

```go
// cmd/command/registry.go
package command

type CommandCategory string

const (
    CategoryBasic     CommandCategory = "Basic"
    CategoryBranch    CommandCategory = "Branch"
    CategoryCommit    CommandCategory = "Commit"
    CategoryRemote    CommandCategory = "Remote"
)

type CommandInfo struct {
    Command     string
    Description string
    Category    CommandCategory
    Usage       string
    Examples    []string
    Subcommands []SubcommandInfo
}

type SubcommandInfo struct {
    Command     string
    Description string
    Usage       string
    Examples    []string
}

var Registry = []CommandInfo{
    {
        Command:     "help",
        Description: "Show help message",
        Category:    CategoryBasic,
        Usage:       "ggc help",
        Examples:    []string{"ggc help"},
    },
    // ... other commands
}
```

### Benefits:

1. **Centralized Management**

   * All command information is maintained in one place.
   * Structured format for storing detailed metadata.

2. **Improved Reusability**

   * All components can reference the registry for consistent information.
   * Used for help generation, interactive mode, and routing.

3. **Better Maintainability**

   * Adding a new command requires editing only the `Registry`.
   * Reduces inconsistency and human error.
   * Enables easy categorization and potential for auto-generating documentation.

4. **Enhanced Extensibility**

   * Easier to add validation or dependency metadata.
   * Flexible enough to add fields like permissions, required env vars, etc.

## Alternative Solutions

1. **Interface-Based Approach**
   Define each command as a type implementing a specific interface.

   * Pros: Stronger type safety
   * Cons: More boilerplate, added complexity

2. **Annotation-Based Approach**
   Use annotations (comments or tags) in handler code to describe metadata.

   * Pros: Closer binding between code and documentation
   * Cons: Limited native support in Go

## Additional Context

This proposal also considers future scalability, such as support for permission checks, dependency declarations, plugin systems, automated documentation generation, and improved testing.

### PR DESCRIPTION
**Pull request by ben-ranford**
## Description of Changes
  - centralised command metadata in cmd/command/ and refactored interactive UI, help system, and routing logic to consume it.
  - added registry validation helpers plus tests covering visibility, cloning, and edge cases.
  - updated contributor guidance for the new workflow (shell completions, doc audit, etc.)

## Related Issue
closes #106 

## Checklist
- [x] I have read the [CONTRIBUTING.md](https://github.com/bmf-san/ggc/blob/main/CONTRIBUTING.md)
- [x] I have added or updated tests
- [x] I have updated the documentation (if required)
- [x] Code is formatted with `make fmt`
- [x] Code passes linter checks via `make lint`
- [x] All tests are passing
- [x] I have run `make docs` to regenerate the README

## Additional Context
  - shell completion scripts still rely on manual updates due to the required format. Could do this as a subsequent issue?
  - ping me if/when this needs a rebase. I will need to rebase the debug work if this gets merged first 😃

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -275,57 +280,105 @@ func (c *Cmd) Route(args []string) {
 
 // routeCommand routes to the appropriate command handler
 func (c *Cmd) routeCommand(cmd string, args []string) {
-	// Handle core commands first
-	if c.handleCoreCommand(cmd, args) {
-		return
+	if c.cmdRouter == nil {
+		c.cmdRouter = mustNewCommandRouter(c)
 	}
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply cmd/cmd.go]: fixed in c208677

File: cmd/templates/help.go [hunk 2]
```diff
@@ -132,6 +99,119 @@ func RenderMainHelp() (string, error) {
...
+		commandregistry.CategoryStatus,
+		commandregistry.CategoryCleanup,
+		commandregistry.CategoryDiff,
+		commandregistry.CategoryTag,
+		commandregistry.CategoryConfig,
+		commandregistry.CategoryHook,
+		commandregistry.CategoryRebase,
+		commandregistry.CategoryStash,
+		commandregistry.CategoryUtility,
+	}
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply cmd/templates/help.go]: fixed in c208677

File: tools/cmd/gendocs/main.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+func generateCommandTable() []string {
+	commands := command.VisibleCommands()
+
+	// Sort commands by category, then by name
+	sort.Slice(commands, func(i, j int) bool {
+		if commands[i].Category != commands[j].Category {
+			return categoryOrder(commands[i].Category) < categoryOrder(commands[j].Category)
+		}
+		return commands[i].Name < commands[j].Name
+	})
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply tools/cmd/gendocs/main.go:143]: fixed in c208677

File: README.md [hunk 4]
```diff
@@ -221,95 +221,96 @@ ggc
```
Comment (bmf-san) [inline README.md:221]: @ben-ranford It looks like this section is currently outputting the Usage instead of the Name of the command. 👀   To keep it consistent with the original README, it would be better to list the available commands by their names. 🙏

**Review by bmf-san — CHANGES_REQUESTED** [review]
@ben-ranford I was hoping to address this issue soon, so this is really helpful—and excellent work! :D  I just have one adjustment I’d like to request, which I’ve left in a comment. 🙏

**Comment by bmf-san** [comment]
> shell completion scripts still rely on manual updates due to the required format. Could do this as a subsequent issue?

Sure, that’s fine. Thanks!

File: README.md (hunk 4 above)
Comment (ben-ranford) [reply README.md:221]: on it! 😄

**Comment by ben-ranford** [comment]
> > shell completion scripts still rely on manual updates due to the required format. Could do this as a subsequent issue?
> 
> Sure, that’s fine. Thanks!

https://github.com/bmf-san/ggc/issues/204 - I'll jump on that once this and the other outstanding PRs I've got are done!

File: README.md (hunk 4 above)
Comment (ben-ranford) [reply README.md:221]: fixed in 797396f 🙂

**Comment by bmf-san** [comment]
@ben-ranford Merged some other PRs earlier — could you fix the conflicts here? 🙏

**Comment by bmf-san** [comment]
LGTM for the most part — I’ll re-review once the conflicts are resolved. :D

**Comment by ben-ranford** [comment]
> @ben-ranford Merged some other PRs earlier — could you fix the conflicts here? 🙏

all done! 🙂

**Review by bmf-san — APPROVED** [review]
LGTM :D This was super helpful, thank you!
